ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -285,31 +280,40 @@ def add_experiences(
                             stored_info.action_masks[idx], padding_value=1
                         )
                     a_dist = stored_take_action_outputs["log_probs"]
+                    # value is a dictionary from name of reward to value estimate of the value head
                     value = stored_take_action_outputs["value"]
                     self.training_buffer[agent_id]["actions"].append(actions[idx])
                     self.training_buffer[agent_id]["prev_action"].append(
                         stored_info.previous_vector_actions[idx]
                     )
                     self.training_buffer[agent_id]["masks"].append(1.0)
-                    if self.use_curiosity:
-                        self.training_buffer[agent_id]["rewards"].append(
-                            next_info.rewards[next_idx] + intrinsic_rewards[next_idx]
-                        )
-                    else:
-                        self.training_buffer[agent_id]["rewards"].append(
-                            next_info.rewards[next_idx]
-                        )
-                    self.training_buffer[agent_id]["action_probs"].append(a_dist[idx])
-                    self.training_buffer[agent_id]["value_estimates"].append(
-                        value[idx][0]
+                    self.training_buffer[agent_id]["done"].append(
+                        next_info.local_done[next_idx]
                     )
-                    if agent_id not in self.cumulative_rewards:
-                        self.cumulative_rewards[agent_id] = 0
-                    self.cumulative_rewards[agent_id] += next_info.rewards[next_idx]
-                    if self.use_curiosity:
-                        if agent_id not in self.intrinsic_rewards:
-                            self.intrinsic_rewards[agent_id] = 0
-                        self.intrinsic_rewards[agent_id] += intrinsic_rewards[next_idx]
+
+                    for name, reward in tmp_rewards_dict.items():
+                        # 0 because we use the scaled reward to train the agent
+                        self.training_buffer[agent_id][
+                            "{}_rewards".format(name)
+                        ].append(tmp_rewards_dict[name][0][next_idx])
```

Comment:
How do you feel about using something like a NamedTuple for the `evaluate` return value? It would make this more obvious.

Reply:
You mean as the unscaled and scaled reward? Something like, e.g. 

in reward_signal.py
```
self.return_value = namedtuple('Reward Signal Result', ['scaled_reward', 'unscaled_reward'])
```

and in the child classes
```
def evaluate():
    ...
    return self.return_value(scaled_reward, unscaled_reward)
```

I agree this would make the `[0]` index more obvious than the way it is now.

Reply:
I guess that works but not very intuitive. You're basically defining a new type:
``` python
RewardSignalResult = namedtuple('RewardSignalResult', ['scaled_reward', 'unscaled_reward'])

...

def evaluate():
    ...
    return RewardSignalResult(scaled_reward, unscaled_reward)

...

... append(tmp_rewards_dict[name].scaled_reward[next_idx])
```

Reply:
OK, I refactored this - let me know what you think